src/world/script_api/7E0E80.c

```diff
@@ -185,7 +194,124 @@ ApiStatus func_80280410(Evt* script, s32 isInitialCall) {
+ApiStatus ShowShopPurchaseDialog(Evt* script, s32 isInitialCall) {
+    PlayerData* playerData = &gPlayerData;
+    GameStatus* gameStatus = gGameStatusPtr;
+    s32 shopItemSlot = script->varTable[0];
+    Shop* shop = gameStatus->mapShop;
+    StaticInventoryItem* shopInventory = &shop->staticInventory[shopItemSlot];
+    StaticItem* shopItem = &gItemTable[shopInventory->itemID];
+    ShopOwner* shopOwner;
+    Evt* shopOwnerScript;
+    EvtSource* shopOwnerScriptSource;
+    s32 bpCost;
+    s32 args;
+
+    shop->flags &= ~SHOP_FLAGS_1;
+    func_800E9900();
+    if (isInitialCall) {
+        D_80286530 = 0;
+        D_80286534 = 0;
+        bpCost = -1;
+        if (shopItem->typeFlags & ITEM_TYPE_FLAGS_40) {
+            bpCost = D_8008F072[shopItem->moveID * 0x14];
+        }
+        script->functionTemp[1] = shop_owner_buy_dialog(0, shopItem->nameMsg, shopInventory->price, bpCost);
+        script->functionTemp[0] = 0;
+        increment_status_menu_disabled();
+        func_800E9900();
+        show_coin_counter();
+    }
+    
+    switch (script->functionTemp[0]) {
+        case 0:
+            if (does_script_exist(script->functionTemp[1]) == 0) {
+                script->functionTemp[0] = 100;
+                script->functionTemp[2] = 0;
+                D_80286528 = msg_get_printer_for_msg(0x1E0001, &script->functionTemp[2]);
+            }
+            break;
+        case 100:
+            args = script->functionTemp[2];
+            if (script->functionTemp[2] == 1) {
+                if (D_80286528->currentOption == 0) {
+                    if (playerData->coins < shopInventory->price) {
+                        script->functionTemp[1] = shop_owner_continue_speech(1);
+                        script->functionTemp[0] = args;
+                    } else if ((shopInventory->itemID - 224) >= 117 && add_item(0) == -1) {
+                        script->functionTemp[1] = shop_owner_continue_speech(2);
+                        script->functionTemp[0] = 4;
+                    } else {
+                        playerData->coins = playerData->coins - shopInventory->price;
+                        if ((shopInventory->itemID - 224) < 117) {
+                            add_badge(shopInventory->itemID);
+                            evt_set_variable(NULL, EVT_SAVE_FLAG(368), 1);
+                        } else {
+                            add_item(shopInventory->itemID);
+                        }
+                        script->functionTemp[1] = shop_owner_continue_speech(3);
+                        script->functionTemp[0] = 5;
+                    }
+                } else {
+                    script->functionTemp[1] = shop_owner_reset_speech(22);
+                    script->functionTemp[0] = 6;
+                }
+            }
+            break;
+        case 1:
+            if (does_script_exist(script->functionTemp[1]) == 0) {
```

Comment:
all these does_script_exist checks are bool checks, so these can be ! instead of == 0